lib/manager/composer/utils.ts

```diff
@@ -69,7 +69,9 @@ export function extractContraints(
   const res: Record<string, string> = { composer: '1.*' };
 
   // extract php
-  if (composerJson.require?.php) {
+  if (composerJson.config?.platform?.php) {
+    res.php = composerJson.config?.platform?.php;
```

Comment:
```suggestion
    res.php = composerJson.config.platform.php;
```

Reply:
Yes, optional chain operator is not required here, as we validate in if.

Reply:
These are all over this file btw, you may wish to enforce this using a lint rule if possible? Do you want me to fix the other lines too? Such as:
```ts
if (composerJson.require?.['composer/composer']) {
    res.composer = composerJson.require?.['composer/composer'];
  } else if (composerJson['require-dev']?.['composer/composer']) {
    res.composer = composerJson['require-dev']?.['composer/composer'];
  }
```